sale_margin_delivered/reports/sale_report.py

```diff
@@ -9,18 +9,9 @@ class SaleReport(models.Model):
 
     margin_delivered = fields.Float(readonly=True)
 
-    def _query(self, with_clause="", fields=None, groupby="", from_clause=""):
-        if fields is None:
-            fields = {}
-        fields.update(
-            {
-                "margin_delivered": " ,SUM(l.margin_delivered /"
-                " COALESCE(s.currency_rate, 1.0)) AS margin_delivered"
-            }
-        )
-        return super()._query(
-            with_clause=with_clause,
-            fields=fields,
-            groupby=groupby,
-            from_clause=from_clause,
-        )
+    def _select_additional_fields(self):
+        res = super()._select_additional_fields()
+        res[
+            "margin_delivered"
+        ] = "SUM(l.margin_delivered / COALESCE(s.currency_rate, 1.0))"
```

Comment:
Follow the guideline for currency conversion in reports: https://github.com/odoo/odoo/blame/116322b705c96a6f23167d09c8f42b569a9e6f22/addons/sale_margin/report/sale_report.py#L15-L16